src/transformers/models/resnet/modeling_resnet.py

```diff
@@ -416,3 +423,84 @@ def forward(
 
         return ImageClassifierOutputWithNoAttention(loss=loss, logits=logits, hidden_states=outputs.hidden_states)
+
+
+@add_start_docstrings(
+    """
+    ResNet backbone, to be used with frameworks like DETR and MaskFormer.
+    """,
+    RESNET_START_DOCSTRING,
+)
+class ResNetBackbone(ResNetPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.resnet = ResNetModel(config)
+
+        # note that this assumes there are always 4 stages
+        self.stage_names = ["stem", "stage1", "stage2", "stage3", "stage4"]
+        self.out_features = config.out_features
+
+        # TODO calculate strides appropriately
+        current_stride = self.resnet.embedder.embedder.convolution.stride[0]
+        self.out_feature_strides = {
+            "stem": current_stride,
+            "stage1": current_stride * 2,
+            "stage2": current_stride * 2,
+            "stage3": current_stride * 2,
+            "stage4": current_stride * 2,
+        }
+
+        self.out_feature_channels = {
+            "stem": config.embedding_size,
+            "stage1": config.hidden_sizes[0],
+            "stage2": config.hidden_sizes[1],
+            "stage3": config.hidden_sizes[2],
+            "stage4": config.hidden_sizes[3],
+        }
+
+        # initialize weights and apply final processing
+        self.post_init()
+
+    @property
+    def channels(self):
+        return [self.out_feature_channels[name] for name in self.out_features]
+
+    @property
+    def strides(self):
+        return [self.out_feature_strides[name] for name in self.out_features]
+
+    @add_start_docstrings_to_model_forward(RESNET_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=BackboneOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(self, pixel_values: Optional[torch.FloatTensor] = None) -> BackboneOutput:
+        """
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import AutoImageProcessor, AutoBackbone
+        >>> import torch
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> processor = AutoImageProcessor.from_pretrained("microsoft/resnet-50")
+        >>> model = AutoBackbone.from_pretrained("microsoft/resnet-50")
+
+        >>> inputs = processor(image, return_tensors="pt")
+
+        >>> outputs = model(**inputs)
+        ```"""
+        outputs = self.resnet(pixel_values, output_hidden_states=True, return_dict=True)
+
+        hidden_states = outputs.hidden_states
+
+        feature_maps = []
+        for idx, stage in enumerate(self.stage_names):
+            if stage in self.out_features:
+                feature_maps.append(hidden_states[idx])
+
+        return BackboneOutput(stage_names=self.out_features, hidden_states=feature_maps)
```

Comment:
`self.out_features` is static, it doesn't change when we change the inputs, so it should just be an attribute of the backbone (as you have done), not included in the `BackboneOutput`.